components/properties/src/script.rs

```diff
@@ -140,6 +171,72 @@ impl<'data> ScriptExtensions<'data> {
             ZeroSlice::from_ule_slice(script_ule_slice)
         }
     }
+
+    /// Returns whether `script` is contained in the Script_Extensions
+    /// property value if the code_point has Script_Extensions, otherwise
+    /// if the code point does not have Script_Extensions then returns
+    /// whether the Script property value matches.
+    ///
+    /// Some characters are commonly used in multiple scripts. For more information,
+    /// see UAX #24: http://www.unicode.org/reports/tr24/.
+    pub fn has_script(&self, code_point: u32, script: Script) -> bool {
+        let sc_with_ext = self.trie.get(code_point);
+
+        if !sc_with_ext.has_extensions() {
+            let script_val = sc_with_ext.0 & SCRIPT_X_SCRIPT_VAL;
```

Comment:
If there are no extensions, then you need not apply the mask, because the upper bits are already 0, right?
(Same in get_script_val())